Since mid october 2021 master branch is producing malformed C++ if thrift file contain comments with double quotes

**Describe the bug**
Since mid october 2021, several downstream projects that contained double quotes failed when compiled with YARP 3.5.100 with errors like:
~~~
Consolidate compiler generated dependencies of target idyntree-yarp-visualizer
[ 29%] Building CXX object src/modules/idyntree-yarp-visualizer/CMakeFiles/idyntree-yarp-visualizer.dir/thrifts/VisualizerCommands.cpp.o
/home/traversaro/test/robotology-superbuild/build/src/idyntree-yarp-tools/src/modules/idyntree-yarp-visualizer/thrifts/VisualizerCommands.cpp:283:43: error: stray ‘\’ in program
  283 |         "@return A status message. \\"[ok]\\" in case of success"
      |                                           ^
/home/traversaro/test/robotology-superbuild/build/src/idyntree-yarp-tools/src/modules/idyntree-yarp-visualizer/thrifts/VisualizerCommands.cpp:283:44: error: stray ‘\’ in program
  283 |         "@return A status message. \\"[ok]\\" in case of success"
      |                                            ^
/home/traversaro/test/robotology-superbuild/build/src/idyntree-yarp-tools/src/modules/idyntree-yarp-visualizer/thrifts/VisualizerCommands.cpp:283:40: error: ‘ok’ was not declared in this scope
  283 |         "@return A status message. \\"[ok]\\" in case of success"
      |                                        ^~
/home/traversaro/test/robotology-superbuild/build/src/idyntree-yarp-tools/src/modules/idyntree-yarp-visualizer/thrifts/VisualizerCommands.cpp:283:45: error: expected ‘}’ before string constant
  283 |         "@return A status message. \\"[ok]\\" in case of success"
      |                                             ^~~~~~~~~~~~~~~~~~~~~
/home/traversaro/test/robotology-superbuild/build/src/idyntree-yarp-tools/src/modules/idyntree-yarp-visualizer/thrifts/VisualizerCommands.cpp:281:40: note: to match this ‘{’
  281 |     static constexpr const char* s_help{
      |                                        ^
/home/traversaro/test/robotology-superbuild/build/src/idyntree-yarp-tools/src/modules/idyntree-yarp-visualizer/thrifts/VisualizerCommands.cpp:283:42: error: expected ‘;’ at end of member declaration
  283 |         "@return A status message. \\"[ok]\\" in case of success"
      |                                          ^
      |                                           ;
/home/traversaro/test/robotology-superbuild/build/src/idyntree-yarp-tools/src/modules/idyntree-yarp-visualizer/thrifts/VisualizerCommands.cpp:283:45: error: expected unqualified-id before string constant
  283 |         "@return A status message. \\"[ok]\\" in case of success"
      |                                             ^~~~~~~~~~~~~~~~~~~~~
/home/traversaro/test/robotology-superbuild/build/src/idyntree-yarp-tools/src/modules/idyntree-yarp-visualizer/thrifts/VisualizerCommands.cpp:285:1: error: expected declaration before ‘}’ token
  285 | };
      | ^
make[5]: *** [src/modules/idyntree-yarp-visualizer/CMakeFiles/idyntree-yarp-visualizer.dir/build.make:126: src/modules/idyntree-yarp-visualizer/CMakeFiles/idyntree-yarp-visualizer.dir/thrifts/VisualizerCommands.cpp.o] Error 1
make[4]: *** [CMakeFiles/Makefile2:377: src/modules/idyntree-yarp-visualizer/CMakeFiles/idyntree-yarp-visualizer.dir/all] Error 2
make[3]: *** [Makefile:136: all] Error 2
make[2]: *** [CMakeFiles/idyntree-yarp-tools.dir/build.make:85: src/idyntree-yarp-tools/CMakeFiles/YCMStamp/idyntree-yarp-tools-build] Error 2
make[1]: *** [CMakeFiles/Makefile2:1819: CMakeFiles/idyntree-yarp-tools.dir/all] Error 2
make: *** [Makefile:101: all] Error 2
~~~

**To Reproduce**
Steps to reproduce the behavior:

Try to configure icub-main `v1.21.0` with YARP 3.5.100 to reproduce the problem. The problem was workarounded in the latest master in https://github.com/robotology/icub-main/pull/767 .


**Expected behavior**
The file generated by thrift should be well-formed.

**Screenshots**
If applicable, add screenshots to help explain your problem.

**Configuration (please complete the following information):**
 - OS:
 - yarp version: 3.5.100
-  compiler: gcc 9

**Additional context**
See https://github.com/robotology/robotology-superbuild/issues/900 . We can also close this issue if this is the intended behavior, I wanted just to have a placeholder/reference for this problem.

--- a/src/yarpidl_thrift/src/t_yarp_generator.cc
+++ b/src/yarpidl_thrift/src/t_yarp_generator.cc
@@ -3720,7 +3720,7 @@ void t_yarp_generator::generate_service_helper_classes_decl(t_function* function
                     }
                     first = false;
                     f_cpp_ << '\n';
-                    f_cpp_ << indent_cpp() << "\"" << replaceInString(helpStr, "\"", "\\\"");
+                    f_cpp_ << indent_cpp() << "\"" << helpStr;
                 }
             }
             indent_down_cpp();

--- a/tests/yarpidl_thrift/demo/demo.thrift
+++ b/tests/yarpidl_thrift/demo/demo.thrift
@@ -13,7 +13,7 @@
 
 
 /**
- * Documentation for enumerator
+ * Documentation for "enumerator"
  */
 enum DemoEnum {
   /** thing 1 */
@@ -24,13 +24,13 @@ enum DemoEnum {
 }
 
 /**
- * Documentation for structure
+ * Documentation for "structure"
  */
 struct DemoStruct {
-  /** this is the x part */
+  /** this is the "x" part */
   1: i32 x = 0,
 
-  /** this is the y part */
+  /** this is the "y" part */
   2: i32 y
 }
 
@@ -43,10 +43,10 @@ struct DemoStructMap {
 }
 
 struct DemoStructExt {
-  /** this is the x part */
+  /** this is the "x" part */
   1: i32 x = 0,
 
-  /** this is the y part */
+  /** this is the "y" part */
   2: i32 y = 0,
 
   /** this is a list of ints */
@@ -68,10 +68,10 @@ struct TestSomeMoreTypes {
 }
 
 /**
- * Documentation for service
+ * Documentation for "service"
  */
 service Demo {
-  /** This function gets the answer */
+  /** This "function" gets the answer */
   i32 get_answer();
   bool set_answer(1:i32 rightAnswer)
   i32 add_one(1:i32 x);

--- a/tests/yarpidl_thrift/demo/main.cpp
+++ b/tests/yarpidl_thrift/demo/main.cpp
@@ -796,7 +796,7 @@ TEST_CASE("IdlThriftTest", "[yarp::idl::thrift]")
         bot.read(con.getReader());
         INFO("Structure specific help is " << bot.toString());
         std::string help = bot.toString();
-        CHECK(help.find("this is the x part") != std::string::npos);
+        CHECK(help.find("this is the \\\"x\\\" part") != std::string::npos);
     }
 
     SECTION("test primitives")

